worker/echo/main.go

```diff
@@ -68,6 +90,17 @@ func echo(
 	return nil
 }
 
+// cancelEcho receives a cancel message id via  com.redhat.Yggdrasil1.Worker1.C.ancel method
+// closes the channel associated to that message to cancel its current run
+func cancelEcho(w *worker.Worker, addr string, id string, cancelID string) error {
+	log.Infof("cancelling message with id %v", cancelID)
+	if cancelChan, exists := sycMapCancelChan.Get(cancelID); exists {
+		close(cancelChan)
```

Comment:
If we've closed the channel because a cancellation was received, the channel
should get deleted from the map too.

Reply:
The channel is deleted from the map when the close message is used, that is in the echo function. The new echo worker functionality may change this.